tensilelite/Tensile/Toolchain/Assembly.py

```diff
@@ -0,0 +1,226 @@
+
+      Args:
+          srcPath: The path to the assembly source file.
+          destPath: The destination path for the generated object file.
+          coVersion: The code object version to use.
+          isa: The target GPU architecture in ISA format.
+          wavefrontSize: The wavefront size to use.
+      """
+      coVersion = getCOVFromParam(coVersion)
+      cpu = getGfxName(isa)
+      wavefrontFlag = "-mwavefrontsize64" if wavefrontSize == 64 else "-mno-wavefrontsize64" 
+    
+      launcher = shlex.split(os.environ.get('Tensile_ASM_COMPILER_LAUNCHER', ''))
+      args = [
+          *launcher, 
+          self.assembler, 
+          "-x", "assembler", 
+          "--target=amdgcn-amd-amdhsa", 
+          f"-mcode-object-version={coVersion}", 
+          f"-mcpu={cpu}",  
+          wavefrontFlag, 
+          "-g" if debug else "",
+          "-c", 
+          "-o", destPath, srcPath
+      ]
+      print2(f"Assembling object files: {' '.join(args)}")
+      subprocess.check_call(args)
+
+    def link(self, srcPaths: List[str], destPath: str):
+        """Links object files into a code object file.
+
+        Args:
+            srcPaths: A list of paths to object files.
+            destPath: A destination path for the generated code object file.
+
+        Raises:
+            RuntimeError: If linker invocation fails.
+        """
+        if os.name == "nt":
+            # Use args file on Windows b/c the command may exceed the limit of 8191 characters
+            with open(Path.cwd() / "clang_args.txt", "wt") as file:
+                file.write(" ".join(objFiles))
+                file.flush()
+            args = [
+                self.assembler,
+                "--target=amdgcn-amd-amdhsa",
+                "-o", destPath, "@clang_args.txt"]
+        else:
+            args = [
+                self.assembler,
+                "--target=amdgcn-amd-amdhsa",
+                "-Xlinker", f"--build-id={self.buildIdKind}",
+                "-o", destPath, *srcPaths
+            ]
+        print2(f"Linking assembly object files into code object: {' '.join(args)}")
+        subprocess.check_call(args)
+
+    def compress(self, srcPath: str, destPath: str, gfx: str):
+        """Compresses a code object file using the provided bundler.
+
+        Args:
+            srcPath: The source path of the code object file to be compressed.
+            destPath: The destination path for the compressed code object file.
+            gfx: The target GPU architecture.
+
+        Raises:
+            RuntimeError: If compressing the code object file fails.
+        """
+        args = [
+            self.bundler,
+            "--compress",
+            "--type=o",
+            "--bundle-align=4096",
+            f"--targets=host-x86_64-unknown-linux,hipv4-amdgcn-amd-amdhsa--{gfx}",
+            "--input=/dev/null",
+            f"--input={srcPath}",
+            f"--output={destPath}",
+        ]
```

Comment:
When we have these things buried in a function they will inevitably change which violates the open/close principle.

Reply:
Okay. Question is then, where does the buck stop? Should I make a parater for `--bundle-align`, `--type`, do I make every parameter configurable? 

By doing this we're pre-emptively guessing about possible future states of our code. Either way, if something changes then we'll have to change the code anyway.

Reply:
yeah it becomes a question of what is application code and what is library code and where do we want stability